cirrus/google_cloud/manager.py

```diff
@@ -1240,6 +1240,65 @@ def _authed_request(self, method, url, data=""):
 
         return response
 
+    def get_project_ancestry(self, project_id=None):
+        """
+        Gets a project's ancestry, represented by a list of
+        resource IDs. The first resource ID is always the
+        project itself, followed by succesive parent resources.
+        https://cloud.google.com/resource-manager/reference/rest/v1/projects/getAncestry
+
+        Args:
+            project_id(str): the project_id of which to get the ancestry,
+                uses self.project_id if None is given
+        Returns:
+            [(str, str)]: a list of tuples, each of which represents a
+            resource ID where the first element of the tuple is the
+            type of the resource ID and the second is the ID itself
+        """
+        project_id = project_id or self.project_id
+        api_url = _get_google_api_url(
+            "projects/" + project_id + ":getAncestry", GOOGLE_CLOUD_RESOURCE_URL
+        )
+        response = self._authed_request("POST", api_url).json()
+
+        ancestors = []
+        if response:
+            response_ancestors = response.get('ancestor')
```

Comment:
to reduce the nesting for future checks like this, early returns are okay. e.g. instead of checking for positive and continuing the nesting, check for negative and early return. e.g. `if not response_ancestors: raise Error()`